### DISCUSSION TITLE
how to encode a yuv422 image to foxglove rawImage proto，is there any example？

### DISCUSSION THREAD

**Discussion by justleap** [foxglove/community#942] [Q&A]
how to encode a yuv422 image to foxglove rawImage proto，is there any example？

**Comment by defunctzombie** [comment]
In which language? There are many languages to pick from.

**Reply by justleap** [reply]
c++

**Reply by defunctzombie** [reply]
Here's an example of writing protobuf mcap files: https://mcap.dev/guides/cpp/protobuf#writing and you can find the RawImage proto here: https://github.com/foxglove/schemas/tree/main/schemas/proto/foxglove

You can put your yuv422 data into this message and set the format to "yuv422"
